File: 3.x.x/crates-utopia/domains/src/registrar/domain.rs
use time::OffsetDateTime;

/// Registrar domain details (PHP `Utopia\Domains\Registrar\Domain`).
#[derive(Debug, Clone, PartialEq, Eq)]
pub struct RegistrarDomain {
    pub domain: String,
    pub created_at: Option<OffsetDateTime>,
    pub expires_at: Option<OffsetDateTime>,
    pub auto_renew: Option<bool>,
    pub nameservers: Option<Vec<String>>,
}

impl RegistrarDomain {
    /// PHP constructor.
    pub fn new(
        domain: impl Into<String>,
        created_at: Option<OffsetDateTime>,
        expires_at: Option<OffsetDateTime>,
        auto_renew: Option<bool>,
        nameservers: Option<Vec<String>>,
    ) -> Self {
        Self {
            domain: domain.into(),
            created_at,
            expires_at,
            auto_renew,
            nameservers,
        }
    }
}

File: packages/domains/src/Domains/Registrar/Domain.php
<?php

declare(strict_types=1);

namespace Utopia\Domains\Registrar;

use DateTime;

final readonly class Domain
{
    public function __construct(
        public string $domain,
        public ?DateTime $createdAt = null,
        public ?DateTime $expiresAt = null,
        public ?bool $autoRenew = null,
        public ?array $nameservers = null,
    ) {}
}
